### ISSUE TITLE
Provide more clarity when stating a coupon code is not available when applied

### ISSUE THREAD

**Issue by polarathene** [gatsbyjs/store.gatsbyjs.org#262]
As per #261 I received an error/notification that was rather vague as to why I could not apply the discount code. 

According to the sidebar(and fact I have not used the code before), it should have been valid to use?

Without further clarity, I am left wondering if it is a bug(in my case it was accepted initially but revoked later during the checkout process), or if it was due to information I had provided during checkout that excluded the availability in some way.

I'm not aware of the conditions for raising the error notice, but if there are cases where it can be made clearer, if they are lengthy, could you add a "More Information" link of some sort?

**Comment by polarathene** [comment]
#113 points out the issue can be due to:

> The email address at checkout needs to exactly match the email you use for GitHub (we link the email to the discount code).

Along with two contact e-mail addresses in that thread. Perhaps it'd be great to let users know when running into this issue as it seems quite a few have and had to search for an answer(I wasn't even aware at the time that the store had a repo here and discovered it via google after a few failed keyword queries). I can see on the main store pages that the footer contains this information, but it's a tad subtle and not present during checkout.

---

**EDIT:** Changing my e-mail address in the checkout process, reapplied the swag code. Order completed, on the sidebar my level 2 code is claimed, but level 1 is still there for some reason(unless it's possible to get that for more than 1 use?)

**Comment by polarathene** [comment]
Had a quick browse over the source, the checkout process is handed off to Shopify by the looks of it so no control regarding that error notification?

Would it be a good idea to provide a hint on the sidebar when displaying the swag codes?

> Enter this discount code during checkout to claim your free swag! ([_having trouble?_](#))

Probably not the right choice of words, but you get the idea :)

If e-mail address is the main/only cause, then just adding the clarity there to use the e-mail address tied to your Github account would be equally helpful. After the unhelpful checkout error, going back to the sidebar with my swag code information was the next place to look for what was going wrong.

**Comment by jlengstorf** [comment]
Yeah, we're not quite sure how best to solve this because we can't control Shopify’s checkout page at all.

We should definitely add more information before the checkout, though. Another option could be an intermediate pop-up as you click checkout that reminds people to use their exact GitHub email address as part of the checkout. Might be a bit annoying, but this trips people up often. 😕

**Comment by polarathene** [comment]
I've not worked with Shopify before, but perhaps reaching out to them might be a good idea as well. Seems like having the ability to add a note like that in their checkout process when needed is a worthwhile feature for them to provide.

A pop-up may not work if user has adblockers that prevent such behaviour?

Also, it seems I somehow got tier 1 used with my personal email that isn't linked to my Github. I have no idea how it worked that time and was rejected the 2nd time around. As a result I ended up having tier 1 unlocked again, this time for my email related to my github account.

**Comment by jlengstorf** [comment]
Hmm... yeah, that seems weird. We'll keep this on the list of things to hopefully dig into later and improve.
